library/src/main/java/com/danikula/videocache/HttpUrlSource.java

```diff
@@ -76,7 +76,11 @@ private int readSourceAvailableBytes(HttpURLConnection connection, int offset, i
     @Override
     public void close() throws ProxyCacheException {
         if (connection != null) {
-            connection.disconnect();
+            try {
+                connection.disconnect();
+            } catch (NullPointerException e) {
```

Comment:
Catching NPE is bad practice. In this case it is just workaround for library bug. So please add some short explanation comments for catch block with link to the [issue](https://github.com/danikula/AndroidVideoCache/issues/29)